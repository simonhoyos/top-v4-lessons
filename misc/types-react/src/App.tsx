import React from 'react';
import logo from './logo.svg';
import './App.css';

interface Props {
  name: string
}

const Greeting: React.FC<Props> = ({ name }) => {
  return (
    <h1>Hola {name}</h1>
  );
}

// interface Props {}
// interface State {}

// class Greeting extends React.Component<Props, State> {

// }

function App() {
  return (
    <div className="App">
      <Greeting name="Simon" />
    </div>
  );
}

export default App;
