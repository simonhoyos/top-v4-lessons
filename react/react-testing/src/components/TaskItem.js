import React from 'react';

class TaskItem extends React.Component {
  render() {
    const { title, done, onClick } = this.props;
    return (
      <div className="task">
        <h2>{title}</h2>
        <p>{done ? 'completada' : 'por completar'}</p>
        <button
          data-testid="complete-task"
          onClick={onClick}
          type="button"
        >
          Completar
        </button>
      </div>
    );
  }
}

export default TaskItem;
