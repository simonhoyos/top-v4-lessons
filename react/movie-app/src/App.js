import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Link,
  Redirect
} from 'react-router-dom';
import Popular from './pages/Popular';
import Upcoming from './pages/Upcoming';
import TopRated from './pages/TopRated';
import NowPlaying from './pages/NowPlaying';
import { MoviesProvider } from './state/MoviesContext';
import './App.css';

function App() {
  return (
    <MoviesProvider>
      <Router>
        <nav>
          <Link to="/popular">Popular</Link>
          <Link to="/upcoming">Upcoming</Link>
          <Link to="/top-rated">Top Rated</Link>
          <Link to="/now-playing">Now Playing</Link>
        </nav>
        <div className="App">
          <Switch>
            <Route exact path="/popular" component={Popular} />
            <Route exact path="/upcoming" component={Upcoming} />
            <Route exact path="/top-rated" component={TopRated} />
            <Route exact path="/now-playing" component={NowPlaying} />
            <Redirect path="*" to="/popular" />
          </Switch>
        </div>
      </Router>
    </MoviesProvider>
  );
}

export default App;
