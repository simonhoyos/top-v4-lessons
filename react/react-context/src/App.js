import React from 'react';
import logo from './logo.svg';
import './App.css';
import TasksContext from './state/TasksContext';

// lifting state
// prop drilling

// Render Props
function FatherComponent({ render }) {
  const [list, setList] = React.useState(['hola mundo', 'hola mir']);
  return render({ list });
}

// class TasksList extends React.Component {
//   static contextType = TasksContext;
//   render() {
//     return (
      // <div className="tasks">
      //   {this.context.tasks.map(task => <p>{task}</p>)}
      // </div>
//     );
//   }
// }

function TasksList() {
  // const { tasks } = React.useContext(TasksContext);

  // return (
    // <div className="tasks">
    //   {tasks && tasks.length > 0 && tasks.map(task => <p>{task}</p>)}
    // </div>
  // );

  return (
    <TasksContext.Consumer>
      {({ tasks }) => (
        <div className="tasks">
          {tasks && tasks.length > 0 && tasks.map(task => <p>{task}</p>)}
        </div>
      )}
    </TasksContext.Consumer>
  );
}

function Button() {
  const { addTask } = React.useContext(TasksContext);

  return <button onClick={addTask}>Agregar tarea</button>;
}

function App() {
  const [tasks, setTasks] = React.useState([]);

  function addTask() {
    setTasks([ ...tasks, 'nueva tarea' ]);
  }

  return (
    <TasksContext.Provider value={{ tasks: ['otro contexto'] }}>
      <TasksList />
      <TasksContext.Provider value={{ tasks, addTask }}>
        <div className="App">
          <TasksList />
          <Button />
          <FatherComponent render={
            ({ list }) => {
              return list.map(el => <p>{el}</p>);
            }
          } />
        </div>
      </TasksContext.Provider>
    </TasksContext.Provider>
  );
}

export default App;
