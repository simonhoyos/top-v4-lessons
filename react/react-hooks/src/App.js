import React from 'react';
import axios from 'axios';
import './App.css';

// class App extends React.Component {
//   state = {
//     count: 0
//   };

//   handleClick = () => {
//     this.setState((prevState) => ({ count: prevState.count + 1 }));
//   }

//   render() {
//     const { count } = this.state;
//     return (
      // <div className="App">
      //   <button onClick={this.handleClick}>{count}</button>
      // </div>
//     );
//   }
// }

function App() {
  const [count, setCount] = React.useState(0);
  const [loading, setLoading] = React.useState(false);
  const [error, setError] = React.useState(null);
  const [posts, setPosts] = React.useState([]);

  React.useEffect(() => {
    setLoading(true);
    axios({
      method: 'GET',
      baseURL: 'http://jsonplaceholder.typicode.com',
      url: '/posts'
    })
      .then(({ data }) => setPosts(data))
      .catch((error) => setError(error))
      .finally(() => setLoading(false));
  }, []);

  function handleClick() {
    setCount(count + 1);
  }

  if(loading) return <p>Loading...</p>;
  if(error) return <p>Something went wrong</p>;
  return (
    <div className="App">
      <button onClick={handleClick}>{count}</button>
      {posts && posts.length && posts.map(({ id, title, body}) => (
        <div key={id} className="post">
          <h2>{title}</h2>
          <p>{body}</p>
        </div>
      ))}
    </div>
  );
}

export default App;
