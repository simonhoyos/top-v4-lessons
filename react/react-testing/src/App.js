import React from 'react';
import uuid from 'uuid-random';
import TasksList from './components/TasksList';
import TaskForm from './components/TaskForm';
import './App.css';

class App extends React.Component {
  state = {
    tasks: [],
    title: '',
  };

  handleChange = e => {
    const { name, value } = e.target;

    this.setState({
      [name]: value,
    });
  }

  handleSubmit = e => {
    e.preventDefault();

    const task = {
      id: uuid(),
      title: this.state.title,
      done: false,
    };

    const tasks = this.state.tasks.concat(task);

    this.setState({
      tasks,
      title: ''
    });
  }

  handleClick = id => {
    return e => {
      const tasks = this.state.tasks.map(task => {
        if(task.id === id) {
          return {
            ...task,
            done: !task.done,
          }
        }

        return task;
      });

      this.setState({ tasks });
    }
  }

  render() {
    const { tasks, title } = this.state;
    return (
      <div className="App">
        <TasksList
          tasks={tasks}
          onClick={this.handleClick}
        />
        <TaskForm
          onChange={this.handleChange}
          onSubmit={this.handleSubmit}
          title={title}
        />
      </div>
    );
  }
}

export default App;
