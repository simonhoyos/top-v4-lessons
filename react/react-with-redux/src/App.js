import React from 'react';
import { connect } from 'react-redux';
import { increaseCount, decreaseCount } from './reducers/countReducer';
import { getPosts } from './reducers/postsReducer';
import './App.css';

function App({
  count,
  posts,
  increaseCount,
  decreaseCount,
  getPosts
}) {
  React.useEffect(() => {
    getPosts()
  }, [getPosts]);

  return (
    <div className="App">
      <h1>{count}</h1>
      <button
        type="button"
        onClick={() => increaseCount(1)}
      >
        Increase
      </button>
      <button
        type="button"
        onClick={decreaseCount}
      >
        Decrease
      </button>
      <div className="posts">
        {posts && posts.length > 0 && posts.map(post => (
          <div className="post" key={post.id}>
            <h2>{post.title}</h2>
            <p>{post.body}</p>
          </div>
        ))}
      </div>
    </div>
  );
}

// HOC - High Order Component
// HOF - High Order Function

const mapStateToProps = ({ countReducer, postsReducer }) => {
  return {
    count: countReducer.count,
    posts: postsReducer.posts
  }
}

const mapDispatchToProps = {
  increaseCount,
  decreaseCount,
  getPosts,
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
