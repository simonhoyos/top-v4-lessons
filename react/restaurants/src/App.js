import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import { Signup } from './pages/Signup';
import { Home } from './pages/Home';
import './App.css';

function PrivateRoute(props) {
  const token = localStorage.token;

  if(!token) return <Redirect to="/signup" />
  return (
    <Route {...props} />
  );
}

function App() {
  return (
    <div className="App">
      <Router>
        <Switch>
          <Route exact path="/admin/signup" component={Signup} />
          <PrivateRoute exact path="/" component={Home} />
          <Redirect from="*" to="/admin/signup" />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
