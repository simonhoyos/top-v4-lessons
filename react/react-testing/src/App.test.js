import React from 'react';
import { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import TaskItem from './components/TaskItem';

describe('App', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<App />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should create a new task', () => {
    const wrapper = mount(<App />);
    const input = wrapper.find('[data-testid="input-title"]');

    input.simulate('change', {
      target: {
        name: 'title',
        value: 'Tarea 1'
      }
    });

    const form = wrapper.find('[data-testid="task-form"]');
    form.simulate('submit');

    // expect(wrapper.state().tasks).toHaveLength(1);
    // expect(wrapper.state().tasks[0].title).toBe('Tarea 1');

    wrapper.update();

    const tasks = wrapper.find(TaskItem);

    expect(tasks).toHaveLength(1);
  });

  it('should complete task', () => {
    const wrapper = mount(<App />);
    const input = wrapper.find('[data-testid="input-title"]');

    input.simulate('change', {
      target: {
        name: 'title',
        value: 'Tarea 1'
      }
    });

    const form = wrapper.find('[data-testid="task-form"]');
    form.simulate('submit');

    input.simulate('change', {
      target: {
        name: 'title',
        value: 'Tarea 2'
      }
    });
    form.simulate('submit');

    wrapper.update();

    const button = wrapper.find('[data-testid="complete-task"]').first();
    button.simulate('click');

    const task = wrapper.find(TaskItem).first();
    expect(task.prop('done')).toBeTruthy();
  });
});
