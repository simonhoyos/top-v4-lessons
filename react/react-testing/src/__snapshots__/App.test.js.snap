// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <TasksList
    onClick={[Function]}
    tasks={Array []}
  />
  <TaskForm
    onChange={[Function]}
    onSubmit={[Function]}
    title=""
  />
</div>
`;
